src/Step/Deploy/SplitDbConnection.php

```diff
@@ -64,43 +60,48 @@ class SplitDbConnection implements StepInterface
     private $configReader;
 
     /**
-     * @var ConfigWriter
+     * @var MagentoShell
      */
-    private $configWriter;
+    private $magentoShell;
 
     /**
-     * @var MagentoShell
+     * @var UpgradeProcess
      */
-    private $magentoShell;
+    private $upgradeRunner;
+
+    /**
+     * @var SlaveConnection
+     */
+    private $slaveConnection;
 
     /**
      * @param DeployInterface $stageConfig
      * @param DbConfig $dbConfig
-     * @param ResourceConfig $resourceConfig
      * @param LoggerInterface $logger
      * @param FlagManager $flagManager
      * @param ConfigReader $configReader
-     * @param ConfigWriter $configWriter
      * @param MagentoShell $magentoShell
+     * @param UpgradeProcess $upgradeRunner
+     * @param SlaveConnection $slaveConnection
      */
     public function __construct(
         DeployInterface $stageConfig,
         DbConfig $dbConfig,
-        ResourceConfig $resourceConfig,
         LoggerInterface $logger,
         FlagManager $flagManager,
         ConfigReader $configReader,
-        ConfigWriter $configWriter,
-        MagentoShell $magentoShell
+        MagentoShell $magentoShell,
+        UpgradeProcess $upgradeRunner,
```

Comment:
Why did you name the class as `UpgradeProcess` but the variable name `$upgradeRunner`.
It is better to rename class to `UpgradeRunner` or variables to `$upgradeProcess` to make naming consistent.

Reply:
My mistake, it was an old class name

Reply:
Done